Epona/include/Epona/Analysis.hpp

```diff
@@ -158,71 +168,126 @@ void FindExtremalVertices(
  *
- *  @attention  There must be at least 7 more elements following given iterator
+ * @param   vertex      the vertex data
+ * @param   translation the vertex translation vector
+ * @param   rotation    the vertex rotation matrix
+ * @param vertex in the world coordinates
+ */
+template < typename Vector, typename Matrix >
+constexpr Vector ModelToWorldSpace(
+        Vector vertex, Vector translation, Matrix rotation
+    )
+{
+    return rotation * vertex + translation;
+}
+
+/**
+ * @brief Calculates vertex position in the model space
  *
- *  @tparam VerticesContainerIt Random access iterator
+ * @tparam  Vector the vector type that has overloaded vector-vector subtraction operator
+ * @tparam  Matrix the matrix type that has overloaded matrix-vector multiplication operator
  *
- *  @param[in]  i       box axis vector
- *  @param[in]  j       box axis vector
- *  @param[in]  k       box axis vector
- *  @param[in]  center  center of the box
- *  @param[out] verticesBeginIterator   iterator to the container
+ * @param   vertex           the vertex data
+ * @param   translation      the vertex translation vector
+ * @param   inverseRotation  the vertex inverse rotation matrix
+ * @param vertex in the model coordinates
  */
-template <typename VerticesContainerIt>
-void CalculateBoxVertices(
-    glm::dvec3 const& i, glm::dvec3 const& j, glm::dvec3 const& k, glm::dvec3 const& center,
-    VerticesContainerIt verticesBeginIterator
-)
+template < typename Vector, typename Matrix >
+constexpr Vector WorldToModelSpace(
+        Vector vertex, Vector translation, Matrix inverseRotation
+    )
 {
-    verticesBeginIterator[0] = (i + j + k) + center;
-    verticesBeginIterator[1] = (i - j + k) + center;
-    verticesBeginIterator[2] = (j - i + k) + center;
-    verticesBeginIterator[3] = (-i - j + k) + center;
-    verticesBeginIterator[4] = (i + j - k) + center;
-    verticesBeginIterator[5] = (i - j - k) + center;
-    verticesBeginIterator[6] = (j - i - k) + center;
-    verticesBeginIterator[7] = (-i - j - k) + center;
+    return inverseRotation * (vertex - translation);
 }
 
 /**
-* @brief Calculates box vertices in the model coordinate space from a given orthogonal basis
-*
-* Writes output vertices to the container starting with @p verticesBeginIterator. There must
-* be at least 7 more elements following given iterator.
-* @tparam VerticesContainerIt Random access iterator
-* @param[in] i box axis vector
-* @param[in] j box axis vector
-* @param[in] k box axis vector
-* @param[out] verticesBeginIterator iterator to the container that is able to store 8 vertices
-*/
-template <typename VerticesContainerIt>
-void CalculateBoxVertices(
-    glm::dvec3 const& i, glm::dvec3 const& j, glm::dvec3 const& k,
-    VerticesContainerIt verticesBeginIterator
-)
+ * @brief  Calculates box vertices in the world coordinate space from a given
+ *         orthogonal basis and it's position
```

Comment:
`it's` is `it is`
if you want to say that something belongs to it you use `its`

Reply:
::)

Reply:
@t0mgit Fixed